src/firebase/dynamicLinks.ts

```diff
@@ -26,8 +26,20 @@ export async function createInviteLink(address: string) {
 }
 
 export async function createJumpstartLink(privateKey: string, networkId: NetworkId) {
-  return dynamicLinks().buildLink({
+  // avoid calling firebase sdk with private key during link creation to protect
+  // the private key from being stored
+  const dynamicLink = await dynamicLinks().buildLink({
     ...commonDynamicLinkParams,
-    link: `${WEB_LINK}jumpstart/${privateKey}/${networkId}`,
+    link: WEB_LINK,
   })
+  const dynamicUrl = new URL(dynamicLink)
+  dynamicUrl.searchParams.set('link', `${WEB_LINK}jumpstart/${privateKey}/${networkId}`)
+
+  // the firebase dynamic links sdk encodes dots and dashes even though it is
+  // not strictly required for urls. calling searchParams.set seems to transform
+  // __all__ search params to be url encoded, where dots and dashes are no
+  // longer encoded. This is probably okay, but to be extra safe we will put
+  // back the encoding ourselves.
+  const searchParams = dynamicUrl.search.replace(/\./g, '%2E').replace(/-/g, '%2D')
+  return `${dynamicUrl.origin}/${searchParams}`
```

Comment:
Could we "not" be extra safe and remove this?
Feels a bit surprising we're only doing it for dots and dashes.

Reply:
i thought about this and tried consuming a link without the encoding on a simulator and it does work, but since the firebase sdk were producing links that are encoded in this way i thought it was safest to keep it incase somewhere in the sdk when consuming deep links on some specific devices, the encoding becomes important :/ since i don't understand why they did the encoding in the first place, i was thinking it wasn't worth the risk of removing it...at least this mess is well documented and inside this function?